sop-extras/src/Data/SOP/Index.hs

```diff
@@ -194,3 +196,8 @@ nsFromIndex n = go 0 sList
       | i == n    = Just $ Z $ K ()
       | otherwise = S <$> go (i + 1) sList
     go !_ SNil    = Nothing
+
+toWord8 :: Index xs x -> Word8
```

Comment:
Smells like a performance bug: strictness would avoid a lot of thunks.

Reply:
Using a strict accumulator